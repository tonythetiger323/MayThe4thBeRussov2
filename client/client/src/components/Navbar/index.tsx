import * as React from 'react'
import Grid from '@material-ui/core/Grid'
import NavbarLink from '../NavbarLink'
import NavbarLinkMenu from '../NavbarLinkMenu'

const Navbar = () => {
  return (
    <Grid
      container
      direction='row'
      alignContent='space-between'
      justify='space-around'
      alignItems='center'
    >
      <Grid item xs={1} />
      <Grid item xs={3}>
        <NavbarLink to='/'>
          <h1>Megan &amp; Randy</h1>
        </NavbarLink>
      </Grid>
      <Grid item xs={8}>
        <NavbarLinkMenu>Their Story</NavbarLinkMenu>
        <NavbarLink to='/ceremony'>Ceremony</NavbarLink>
        <NavbarLink to='/reception'>Reception</NavbarLink>
        <NavbarLink to='/registry'>Registry</NavbarLink>
        <NavbarLink to='/guestbook'>Guestbook</NavbarLink>
        <NavbarLink to='/contactus'>Contact Us</NavbarLink>
      </Grid>
    </Grid>
  );
};
export default Navbar;
