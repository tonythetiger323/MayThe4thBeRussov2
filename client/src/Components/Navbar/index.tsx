import * as React from 'react'
import Grid from '@material-ui/core/Grid'
import NavbarLink from '../NavbarLink'
import NavbarLinkMenu from '../NavbarLinkMenu'

const Navbar = () => {
  return (
    <Grid item container alignItems='center'>
      <Grid item>
        <NavbarLinkMenu>Their Story</NavbarLinkMenu>
        <NavbarLink to='/ceremony'>Ceremony</NavbarLink>
        <NavbarLink to='/reception'>Reception</NavbarLink>
        <NavbarLink to='/rsvp'>RSVP</NavbarLink>
        <NavbarLink to='/registry'>Registry</NavbarLink>
        <NavbarLink to='/guestbook'>Guestbook</NavbarLink>
        <NavbarLink to='/contactus'>Contact Us</NavbarLink>
      </Grid>
    </Grid>
  )
}
export default Navbar
