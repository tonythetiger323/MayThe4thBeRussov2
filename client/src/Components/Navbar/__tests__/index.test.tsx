import * as React from 'react'
import { shallow } from 'enzyme'
import Navbar from '../../Navbar'
import { createShallow } from '@material-ui/core/test-utils'
import Grid from '@material-ui/core/Grid'

let wrapper: any
let shallowMui: any

describe('Navbar Component', () => {
  beforeEach(() => (wrapper = shallow(<Navbar />)))

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('<Grid/>', () => {
  beforeEach(() => {
    shallowMui = createShallow()
  })

  it('shoule work', () => {
    wrapper = shallowMui(<Grid />)
  })
})
