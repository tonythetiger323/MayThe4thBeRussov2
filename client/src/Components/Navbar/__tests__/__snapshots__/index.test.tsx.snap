// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": <WithStyles(Grid)
        item={true}
      >
        <NavbarLink
          to="/"
        >
          Home
        </NavbarLink>
        <NavbarLink
          to="/theirstory"
        >
          Their Story
        </NavbarLink>
        <NavbarLink
          to="/ceremony"
        >
          Ceremony
        </NavbarLink>
        <NavbarLink
          to="/reception"
        >
          Reception
        </NavbarLink>
        <NavbarLink
          to="/rsvp"
        >
          RSVP
        </NavbarLink>
        <NavbarLink
          to="/registry"
        >
          Registry
        </NavbarLink>
        <NavbarLink
          to="/guestbook"
        >
          Guest Book
        </NavbarLink>
      </WithStyles(Grid)>,
      "container": true,
      "item": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <NavbarLink
            to="/"
          >
            Home
          </NavbarLink>,
          <NavbarLink
            to="/theirstory"
          >
            Their Story
          </NavbarLink>,
          <NavbarLink
            to="/ceremony"
          >
            Ceremony
          </NavbarLink>,
          <NavbarLink
            to="/reception"
          >
            Reception
          </NavbarLink>,
          <NavbarLink
            to="/rsvp"
          >
            RSVP
          </NavbarLink>,
          <NavbarLink
            to="/registry"
          >
            Registry
          </NavbarLink>,
          <NavbarLink
            to="/guestbook"
          >
            Guest Book
          </NavbarLink>,
        ],
        "item": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Their Story",
            "to": "/theirstory",
          },
          "ref": null,
          "rendered": "Their Story",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Ceremony",
            "to": "/ceremony",
          },
          "ref": null,
          "rendered": "Ceremony",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Reception",
            "to": "/reception",
          },
          "ref": null,
          "rendered": "Reception",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "RSVP",
            "to": "/rsvp",
          },
          "ref": null,
          "rendered": "RSVP",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Registry",
            "to": "/registry",
          },
          "ref": null,
          "rendered": "Registry",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Guest Book",
            "to": "/guestbook",
          },
          "ref": null,
          "rendered": "Guest Book",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": <WithStyles(Grid)
          item={true}
        >
          <NavbarLink
            to="/"
          >
            Home
          </NavbarLink>
          <NavbarLink
            to="/theirstory"
          >
            Their Story
          </NavbarLink>
          <NavbarLink
            to="/ceremony"
          >
            Ceremony
          </NavbarLink>
          <NavbarLink
            to="/reception"
          >
            Reception
          </NavbarLink>
          <NavbarLink
            to="/rsvp"
          >
            RSVP
          </NavbarLink>
          <NavbarLink
            to="/registry"
          >
            Registry
          </NavbarLink>
          <NavbarLink
            to="/guestbook"
          >
            Guest Book
          </NavbarLink>
        </WithStyles(Grid)>,
        "container": true,
        "item": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <NavbarLink
              to="/"
            >
              Home
            </NavbarLink>,
            <NavbarLink
              to="/theirstory"
            >
              Their Story
            </NavbarLink>,
            <NavbarLink
              to="/ceremony"
            >
              Ceremony
            </NavbarLink>,
            <NavbarLink
              to="/reception"
            >
              Reception
            </NavbarLink>,
            <NavbarLink
              to="/rsvp"
            >
              RSVP
            </NavbarLink>,
            <NavbarLink
              to="/registry"
            >
              Registry
            </NavbarLink>,
            <NavbarLink
              to="/guestbook"
            >
              Guest Book
            </NavbarLink>,
          ],
          "item": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Their Story",
              "to": "/theirstory",
            },
            "ref": null,
            "rendered": "Their Story",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Ceremony",
              "to": "/ceremony",
            },
            "ref": null,
            "rendered": "Ceremony",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Reception",
              "to": "/reception",
            },
            "ref": null,
            "rendered": "Reception",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "RSVP",
              "to": "/rsvp",
            },
            "ref": null,
            "rendered": "RSVP",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Registry",
              "to": "/registry",
            },
            "ref": null,
            "rendered": "Registry",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Guest Book",
              "to": "/guestbook",
            },
            "ref": null,
            "rendered": "Guest Book",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
